### PR TITLE
[BH-1814] Fix EInk errors in logs

### PR DESCRIPTION
**Pull request by dawidwojtas-mudita**
According to the newest fsl library, the LPSPI_Reset function shouldn't be invoked in the LPSPI_MasterInit function.
As a result, the LPSPI module can't work after calling this API.

Fix the system crash when an EInk error occurs. If the display returns the error the driver could invoke the EinkPowerDown function recursively which causes a crash.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

**Review by MateuszSz-mudita — APPROVED** [review]

**Review by lukasmas — APPROVED** [review]

**Review by Maciej-Mudita — APPROVED** [review]

File: module-bsp/board/rt1051/bsp/eink/bsp_eink.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ typedef struct _bsp_eink_driver
 namespace
 {
     using namespace drivers;
-    static lpspi_master_config_t s_eink_lpspi_master_config;
+    static lpspi_master_config_t spi_config;
     static std::shared_ptr<drivers::DriverGPIO> gpio;
```
Comment (Lefucjusz) [inline module-bsp/board/rt1051/bsp/eink/bsp_eink.cpp:83]: `static` seems to be redundant here, as all of these objects are declared in anonymous namespace, also `using namespace drivers` is not needed if I understand correctly, as references to classes are done with `drivers::` anyway

File: module-bsp/board/rt1051/bsp/eink/bsp_eink.cpp [hunk 2]
```diff
@@ -188,25 +186,24 @@ status_t BSP_EinkInit()
...
+    LPSPI_MasterGetDefaultConfig(&spi_config);
 
-    s_eink_lpspi_master_config.whichPcs           = kLPSPI_Pcs0;
-    s_eink_lpspi_master_config.pcsActiveHighOrLow = kLPSPI_PcsActiveLow;
+    spi_config.baudRate                      = BSP_EINK_TRANSFER_WRITE_CLOCK;
+    spi_config.bitsPerFrame                  = 8;
+    spi_config.cpol                          = kLPSPI_ClockPolarityActiveHigh;
+    spi_config.cpha                          = kLPSPI_ClockPhaseFirstEdge;
+    spi_config.direction                     = kLPSPI_MsbFirst;
+    spi_config.pcsToSckDelayInNanoSec        = 1000000000 / spi_config.baudRate;
```
Comment (Lefucjusz) [inline module-bsp/board/rt1051/bsp/eink/bsp_eink.cpp:196]: Maybe some constexpr for that value? I guess it's number of nanoseconds in second :smile:

**Review by Lefucjusz — APPROVED** [review]

File: module-bsp/board/rt1051/bsp/eink/bsp_eink.cpp (hunk 1 above)
Comment (dawidwojtas-mudita) [reply module-bsp/board/rt1051/bsp/eink/bsp_eink.cpp]: Thanks, done ;)

File: module-bsp/board/rt1051/bsp/eink/bsp_eink.cpp (hunk 2 above)
Comment (dawidwojtas-mudita) [reply module-bsp/board/rt1051/bsp/eink/bsp_eink.cpp]: It is just copied from the fsl library, but I agree. Added constexpr value ;)

**Pull request merged**